### PR TITLE
Add wstETH Arbitrum

### PR DESCRIPTION
**Pull request by efecarranza**
# Changelog

Add wstETH Arbitrum listing.
Add tests.
Add diff reports.

### CODE REVIEW

File: src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * @title List wstETH on AaveV3Arbitrum
+ * @author Llama
+ * @dev This proposal lists wstETH on Aave V3 Arbitrum
+ * Governance: https://governance.aave.com/t/arc-add-support-for-wsteth-on-arbitrum-aave-v3/11387/11
+ * Snapshot: https://snapshot.org/#/aave.eth/proposal/0x402647b83c436aecbe2fe404870f08767b5509225cbe606913e50801f87f5db8
+ */
+contract AaveV3ArbWSTETHListingPayload is AaveV3ListingArbitrum {
+  address public constant WSTETH = 0x5979D7b546E38E414F7E9822514be443A4800529;
+  address public constant WST_ETH_PRICE_FEED = 0x230E0321Cf38F09e247e50Afc7801EA2351fe56F;
```
Comment (0xrajath) [inline src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol:16]: I couldn't find this price feed on the Chainlink Documentation. Where did you get it from?

File: src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    listings[0] = IGenericV3ListingEngine.Listing({
+      asset: WSTETH,
+      assetSymbol: 'wstETH',
+      priceFeed: WST_ETH_PRICE_FEED,
+      rateStrategy: INTEREST_RATE_STRATEGY,
+      enabledToBorrow: true,
+      stableRateModeEnabled: false,
+      borrowableInIsolation: false,
+      withSiloedBorrowing: false,
```
Comment (0xrajath) [inline src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol:47]: Just a general question. According to https://governance.aave.com/t/arc-add-support-for-wsteth-on-arbitrum-aave-v3/11387/7 , Borrowable is `Yes` and Collateral Enabled is `Yes`. How do those 2 things relate to the parameters here?

File: src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  string public constant EMODE_LABEL_ETH_CORRELATED = 'ETH correlated';
+  uint16 public constant EMODE_LTV_ETH_CORRELATED = 90_00;
+  uint16 public constant EMODE_LT_ETH_CORRELATED = 93_00;
+  uint16 public constant EMODE_LBONUS_ETH_CORRELATED = 10_200;
+  uint8 public constant EMODE_CATEGORY_ID_ETH_CORRELATED = 2;
+
+  constructor() AaveV3ListingArbitrum(IGenericV3ListingEngine(AaveV3Arbitrum.LISTING_ENGINE)) {}
+
+  function getAllConfigs() public override returns (IGenericV3ListingEngine.Listing[] memory) {
```
Comment (0xrajath) [inline src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol:27]: Another general question. I noticed that it's not an `execute()` function here. Hope you'll be handling that in the proposal script execution appropriately.

File: src/test/arbitrum/AaveV3ArbWSTETHListingPayloadTest.t.sol [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+pragma solidity 0.8.17;
+
+import 'forge-std/Test.sol';
+import {AaveV3Arbitrum} from 'aave-address-book/AaveAddressBook.sol';
+import {ProtocolV3TestBase, InterestStrategyValues, ReserveConfig} from 'aave-helpers/ProtocolV3TestBase.sol';
+import {AaveGovernanceV2} from 'aave-address-book/AaveGovernanceV2.sol';
+import {TestWithExecutor} from 'aave-helpers/GovHelpers.sol';
+import {AaveV3ArbWSTETHListingPayload} from '../../contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol';
+
+contract AaveV3ArbWSTETHListingPayloadTest is ProtocolV3TestBase, TestWithExecutor {
```
Comment (0xrajath) [inline src/test/arbitrum/AaveV3ArbWSTETHListingPayloadTest.t.sol:11]: Do you think we might need some post execution tests, checking that you can supply this as collateral and also borrow against it? Does BGD recommend that?

**Review by 0xrajath — APPROVED** [review]
LGTM. Left some comments

File: src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol (hunk 1 above)
Comment (efecarranza) [reply src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol:16]: Shared via DM.

File: src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol (hunk 2 above)
Comment (efecarranza) [reply src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol:47]: Hi,

`enabledToBorrow` is `BORROWABLE` on the post.
If `liqThreshold` is 0, then collaterable is not enabled. If you have a `liqThreshold`, it is enabled.

File: src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol (hunk 3 above)
Comment (efecarranza) [reply src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol:27]: That's been abstracted away by BGD here: https://github.com/bgd-labs/aave-helpers/blob/master/src/v3-listing-engine/AaveV3ListingBase.sol#L29

**Review by rustboyar — APPROVED** [review]

File: src/test/arbitrum/AaveV3ArbWSTETHListingPayloadTest.t.sol (hunk 4 above)
Comment (sakulstra) [reply src/test/arbitrum/AaveV3ArbWSTETHListingPayloadTest.t.sol:12]: we have a e2e suite, but for "normal"(no custom a/s/v implementations) listing i think configuration snapshot it enough. We opt to not test the protocol standard functionality (as the protocol is tested heavily).

File: src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    AaveV3Arbitrum.POOL_CONFIGURATOR.setEModeCategory(
```
Comment (rustboyar) [inline src/contracts/arbitrum/AaveV3ArbWSTETHListingPayload.sol:28]: It is better to create eMode in the `_preExecute()` hook. Can't find an example, but basically you can override `_preExecute()` here an create eMode category there.

It is a bit strange to have it in `getAllConfigs()` (see `AaveV3ListinBase.sol`)

**Comment by rustboyar** [comment]
Should WETH also be added to this eMode? It makes not much sense to have only one asset in the eMode category

**Review by rustboyar — APPROVED** [review]

**Pull request merged**